### DISCUSSION TITLE
Towards 2.12 release

### DISCUSSION THREAD

**Discussion by scop** [scop/bash-completion#530] [General]
I'd like to get 2.12 out soon, mainly due to ca361be02a92e30a3d26d376b30faf268853e836 and #506 so that we don't forget to add a prominent note about it to the release notes :)

The only in-progress things I'd like to complete before that is fixing the test suite (#529 will likely do that), and to enable macOS in GH actions tests. The `macos-ci` branch works towards that -- it's a mess currently, I'll do some cleanups -- but there are some failures to fix: https://github.com/scop/bash-completion/runs/2688724384?check_suite_focus=true
- a bunch of `make` failures to which I haven't had a closer look, maybe the fix would be just to tolerate nonexistence of the file we're trying to remove, or maybe we should be running that in a tmpdir that we can write to
- a `man` failure in `test_4`

**Comment by scop** [comment]
Oh, and releases config with GH workflows should be done before we can release, TODO is in .github/workflows/check.yaml. Maybe simples to just do the release from there, as distcheck with which we test builds the tarball we can release while at it.

**Reply by scop** [reply]
Should also note that I don't have a macOS box easily available for hacking this, so I've been working based on the CI output.

**Reply by akinomyoga** [reply]
Unfortunately, I don't have an access to macOS either.

**Comment by scop** [comment]
I've set up https://github.com/scop/bash-completion/projects/2 for remaining items I'd like to see for 2.12.

Situation with macOS tests is quite uncertain at this point, I still only have the CI to "conveniently" work with, and the remaining failures are kind of flaky. Therefore I'm fine with going ahead with releasing 2.12 without this taken care of, but wouldn't mind if we happen to get it towrking. Anyway when run separately, the failure cases pass, but when we run the whole suite, they fail. Maybe something's not cleaned up by preceding tests, dunno. Here's one whole run in which make and ps tests fail:
https://github.com/scop/bash-completion/runs/3702484291?check_suite_focus=true
...and here's one where one of the failing make tests is run separately, and it succeeds that way:
https://github.com/scop/bash-completion/runs/3709618620?check_suite_focus=true

So in essence I'd like us to decide on the naming things and implement accordingly, and finally apply those renovate dependency bumps and we're good for release.

**Comment by inconstante** [comment]
Do we still have plans to release 2.12 once the tasks in projects/2 are done?

**Reply by scop** [reply]
As far as I'm concerned, yes.

**Comment by scop** [comment]
@akinomyoga you used to have a pile of WIP branches related to things on the 2.12 TODO list. What's the status as far as you're concerned, do you have some related unsubmitted work, or do you feel we're blocked waiting for something before we can proceed with the remaining TODO items?

I don't have a crystal clear picture, this has been on hold again for too long again (all my bad), but I'd like to work towards getting the remaining items done, without ignoring/stomping on any existing works/concerns there might be.

**Reply by akinomyoga** [reply]
I have some WIP branches for switching to `_comp_split`, but I think that those do not necessarily need to be a part of the 2.12 release. What I think needs to be finished for 2.12 is the API change including the function names and their uses of arguments/shell variables. Today I'm busy, so I try to reply in more detail tomorrow.

**Reply by scop** [reply]
The 2.12 project is now clear for release as far as I'm concerned, will let things sit for a bit and roll the release soon unless something important pops up.

We can continue to merge smaller PR's still before the release, but anything more disruptive should be held back.

**Reply by akinomyoga** [reply]
I agree with releasing 2.12 in the current state.

**Comment by akinomyoga** [comment]
I think this can be closed since 2.12 has been released and there's no remaining issue (if #506 is now OK).
